src/flowcean/transforms/match_sampling_rate.py

```diff
@@ -13,99 +13,166 @@ class MatchSamplingRate(Transform):
+                - 'linear'
+                - 'slerp' (to be implemented)
         """
-        self.reference_timestamps = reference_timestamps
-        self.feature_columns_with_timestamps = feature_columns_with_timestamps
+        self.reference_feature = reference_feature
+        self.feature_columns = feature_columns
 
     def transform(self, data: pl.DataFrame) -> pl.DataFrame:
+        """Transform the input DataFrame.
+
+        Args:
+            data: Input DataFrame.
+
+        Returns:
+            Transformed DataFrame.
+
+        """
+        # preserve all constant columns that are not timeseries data
+        non_timeseries_features = [
+            col
+            for col in data.columns
+            if col not in [*self.feature_columns, self.reference_feature]
+        ]
         logger.debug("Matching sampling rate of time series.")
 
-        for i in range(len(data)):
-            reference_timestamps = data[self.reference_timestamps][i]
-            for (
-                feature,
-                timestamp,
-            ) in self.feature_columns_with_timestamps.items():
-                timestamps = data[timestamp][i].to_numpy()
-                feature_data = data[feature][i].to_numpy()
-                resampled_timeseries = interp(
-                    reference_timestamps,
-                    timestamps,
-                    feature_data,
-                ).tolist()
-                data = (
-                    data.lazy()
-                    .with_row_index()
-                    .with_columns(
-                        pl.when(pl.col("index") == i)
-                        .then(resampled_timeseries)
-                        .otherwise(pl.col(feature))
-                        .alias(feature)
-                    )
-                    .drop("index")
-                    .collect()
+        if self.reference_feature not in data.columns:
+            msg = f"Reference {self.reference_feature} not found in the DataFrame."  # noqa: E501
+            raise ValueError(msg)
+
+        features = list(self.feature_columns.keys())
+        reference_df = (
+            data.select(pl.col(self.reference_feature).explode())
+            .unnest(cs.all())
+            .rename(
+                lambda name: self.reference_feature + "_" + name
+                if name != "time"
+                else name
+            )
+        )
+        result = pl.concat(
+            [
+                self.interpolate_feature(feature, data, reference_df)
+                for feature in features
+            ],
+            how="horizontal",
+        )
+
+        return pl.concat(
+            [
+                data.select(self.reference_feature),
+                result,
+                data.select(non_timeseries_features),
+            ],
+            how="horizontal",
+        )
+
+    def interpolate_feature(
+        self, feature: str, data: pl.DataFrame, reference_df: pl.DataFrame
```

Comment:
Do not repeat yourself (DRY)
```suggestion
        self, feature: str, data: pl.DataFrame, reference: pl.DataFrame
```
Do not mention types in variable names

Reply:
renamed the variable